src/res_printer.ml

```diff
@@ -4818,17 +4822,28 @@ and printDirectionFlag flag = match flag with
   | Asttypes.Downto -> Doc.text " downto "
   | Asttypes.Upto -> Doc.text " to "
 
-and printRecordRow (lbl, expr) cmtTbl =
+and printRecordRow (lbl, expr) cmtTbl punningAllowed =
   let cmtLoc = {lbl.loc with loc_end = expr.pexp_loc.loc_end} in
-  let doc = Doc.group (Doc.concat [
-    printLidentPath lbl cmtTbl;
-    Doc.text ": ";
-    (let doc = printExpressionWithComments expr cmtTbl in
-    match Parens.expr expr with
-    | Parens.Parenthesized -> addParens doc
-    | Braced braces -> printBraces doc expr braces
-    | Nothing -> doc);
-  ]) in
+  let doc = Doc.group (
+    match expr.pexp_desc with
+    | Pexp_ident({txt = Lident key; loc = keyLoc}) when (
+      punningAllowed && 
+      Longident.last lbl.txt = key &&
```

Comment:
Do we want to print `A.a: a` punned? I always found this super unclear, but this is probably my personal opinion?

Reply:
Yes, I would like to be able to print just `A.a`. But there is currently no "auto-punning" implemented - if the input is `A.a: a`, it will be kept that way.